### PR TITLE
py-pyobjc: update to 10.1, remove py37, support 10.9

### PR DESCRIPTION
**Pull request by erikbs**
#### Description

Updating PyObjC to version 10.1. The developer has made a great effort to support older OS versions, so that OS X 10.9 and later are now supported (down from 10.13).

This version requires Python 3.8 or later, so I removed the py37 subport. I considered pinning 3.7 to the older version, but judging by other commits to master, many py37 subports are now being removed. I have also removed the `py37` subport of four ports that depend on this one: `py37-imagecodecs`, `py37-tensorflow-addons`, `py37-zope-component`, `py37-zope-hookable`.

Changes:
* Blacklist Clang < 900 because they do not support the option `-flto=thin`
* Require full Xcode (`use_xcode yes`) because `xcodebuild` complained that it could not use CommandLineTools (happened both locally on OS X 10.9 and on the macOS 11.0 CI bot)
* 10.9 only: use the OS X 10.10 SDK instead of setting SDK root to `/`. Needed because 10.9 lacks the needed SIMD headers. The 10.10 SDK is included with the latest Xcode versions that 10.9 supports. Solution inspired by `py-lightblue`.
* Updated line numbers in patch, removed patch for `IMServicePlugIn` framework, which no longer exists

###### Type(s)

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on
macOS 10.9.5 13F1911 x86_64
Xcode 6.2 6C131e

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?

Notes:
* `lint` gives one warning:
  `Warning: Line 85 seems to hardcode the version number, consider using ${version} instead`
  This is the line where I hardcoded the use of MacOSX10.10.sdk on Darwin 13.
* There are no tests enabled
* There are no binaries to test
* I tested (on Python 3.12) by importing `objc` and calling a couple of utility functions, as well as by building a port that depends on this one and tested it

### CODE REVIEW

File: python/py-pyobjc/Portfile
Comment (reneeotten) [inline python/py-pyobjc/Portfile:33]: You cannot just remove the py37 subport as there are still other ports that depend on it:
```
❯ port echo depends:py37-pyobjc
py37-imagecodecs
py37-zope-hookable
py37-tensorflow-addons
```

**Review by reneeotten — CHANGES_REQUESTED** [review]
thanks for the update @erikbs ; one comment regarding the proposed removal of the py37 subport

File: python/py-pyobjc/Portfile
Comment (erikbs) [reply python/py-pyobjc/Portfile:33]: Good point! I will make a separate PR on removing the py37 subports first then. Converting this one to draft in the meanwhile

Comment (reneeotten) [reply python/py-pyobjc/Portfile:33]: that change should be made in this PR please. Also, please note that this has been done recursively - so you'll need to check if any of the three ports listed above still have dependencies, and, if so, continue that process until there isn't anything showing up anymore.

Comment (erikbs) [reply python/py-pyobjc/Portfile:33]: Got it :) What do I do about the PR title / commit message title – should I mention those other ports as well or is it sufficient to mention them in the body of the commit message?

**Comment by erikbs** [comment]
One more thing that I forgot: even though I lowered the Darwin version in `platforms`, I still get a warning about this being a port that is known to fail on my system. Do I have to edit something else to remove it or will this go automatically? I found no `known_fail` statement in the Portfile. Log:
```
install --unrequested py312-pyobjc
py312-pyobjc is known to fail. Try to install anyway? [y/N]: y
--->  Computing dependencies for py312-pyobjc
--->  Fetching archive for py312-pyobjc
--->  Attempting to fetch py312-pyobjc-10.1_0.darwin_13.x86_64.tbz2 from http://packages.macports.org/py312-pyobjc
--->  Attempting to fetch py312-pyobjc-10.1_0.darwin_13.x86_64.tbz2 from http://cph.dk.packages.macports.org/py312-pyobjc
--->  Attempting to fetch py312-pyobjc-10.1_0.darwin_13.x86_64.tbz2 from http://fra.de.packages.macports.org/py312-pyobjc
--->  Fetching distfiles for py312-pyobjc
--->  Verifying checksums for py312-pyobjc
--->  Extracting py312-pyobjc
--->  Applying patches to py312-pyobjc
--->  Configuring py312-pyobjc
--->  Building py312-pyobjc
--->  Staging py312-pyobjc into destroot
--->  Installing py312-pyobjc @10.1_0
--->  Activating py312-pyobjc @10.1_0
--->  Cleaning py312-pyobjc
--->  Scanning binaries for linking errors
--->  No broken files found.
--->  No broken ports found.
```

File: python/py-pyobjc/Portfile
Comment (erikbs) [reply python/py-pyobjc/Portfile:33]: CI failed after I touched `py-tensorflow-addons` because its py38 subport (which I did not remove at first) depended on now-removed `py38-tensorflow`, so I removed that subport and its (only) dependent, `py38-tensor2tensor`.

Comment (reneeotten) [reply python/py-pyobjc/Portfile:33]: these should be in different commits but in the same PR. 
Also, why did you remove PY38 subports; that Python version is still supported so they should remain.

Comment (erikbs) [reply python/py-pyobjc/Portfile:33]: Okay, I will redo the commits.

I can put back the py38 subports, but they will still not work since `py38-tensorflow` does not exist anymore ([dropped in May](https://github.com/macports/macports-ports/commit/fb29cdd5712878122fe6284407d0d977a1194154)).

Comment (reneeotten) [reply python/py-pyobjc/Portfile:33]: oh, I see - that was already broken indeed. Yes, sure just remove the py38 subports then; please refer to the commit that incorrectly dropped `py38-tensorflow` in that commit message.

Comment (erikbs) [reply python/py-pyobjc/Portfile:33]: I just double-checked the dependency on `py38-tensorflow` in the `py-tensorflow-addons` Portfile before removing `38` and I am unsure this is a hard dependency after all. There is this line (a build dependency):

`path:${python.pkgd}/tensorflow:py${python.version}-tensorflow`

If I have understood correctly, the `path:` thing means that one could meet the dependency by installing `tensorflow` from somewhere else, for example an old version from PIP. There is also a port `py-tensorflow1` that does have a 38 subport. It is incompatible with my system, but I guess there is a good chance it will install to `${python.pkgd}`, because it conflicts with the one without the `1` suffix. I am trying to push a commit that replaces this dependency line with the following:

```
if {${python.version} eq 38} {
    depends_build-append \
        path:${python.pkgd}/tensorflow:py${python.version}-tensorflow1
} else {
    depends_build-append \
        path:${python.pkgd}/tensorflow:py${python.version}-tensorflow
}
```
If that passes CI I guess I can keep the py38 subports?

Comment (reneeotten) [reply python/py-pyobjc/Portfile:33]: just leave the py38 subport; it's not something you changed in this PR so it's up to the maintainer to fix a possible problem. The fact that the CI will not pass because of that is fine. Please change as little as possible that is needed for main change in the PR to work (i.e., the update to the latest upstream version requires to drop the PY37 subport and thus dependents as well).

**Comment by jmroot** [comment]
> One more thing that I forgot: even though I lowered the Darwin version in `platforms`, I still get a warning about this being a port that is known to fail on my system. Do I have to edit something else to remove it or will this go automatically? I found no `known_fail` statement in the Portfile.

Sounds like a stale PortIndex.

**Comment by pmetzger** [comment]
Is this ready to commit?

**Comment by erikbs** [comment]
Ready, just waiting for approval

**Comment by pmetzger** [comment]
@erikbs Approval from whom?

**Comment by erikbs** [comment]
@pmetzger from any reviewer (action box says there is one pending reviewer), I just need that someone takes a final look to verify that the changes look okay

**Review by danchr — APPROVED** [review]
PyObjC is as much a scripting environment as it is a library, so normally, I'd request that you retain the old Python version as a separate port. While the binary compatibility is generally good, older systems occasionally fail to build.

However, since you're actually _lowering_ the requirement here, this is fine 🙂

**Comment by barracuda156** [comment]
From the version history it seems that older macOS were supported. We can add a fallback for pre-10.9 down to 10.5. Let me ask the developer which version would he recommend to use.

**Comment by reneeotten** [comment]
@erikbs please resolve the conflict(s) and update the message for the second commit to follow the guidelines. 

It should be "py-{portA,portB,...}: drop py37 subport"

**Comment by erikbs** [comment]
I tried `py-{tensorflow-addons,zope-{component,hookable}}: drop py37 subport`, but at 68 characters it is above the hard limit of 60. Since tensorflow-addons and the other two are completely unrelated, I split that commit in two.

**Pull request merged**
